### DISCUSSION TITLE
execute script on game open/close

### DISCUSSION THREAD

**Discussion by wieluk** [Phalcode/gamevault-app#396] [General]
Hey I just wanted to ask if it is possible to execute a script before a games is opend and after it is closed?

**Comment by Alfagun74** [comment]
Yes you can just write some batch script and select that as your launch exe if you want

**Reply by wieluk** [reply]
So I have to do it for every single game individually?

**Reply by Alfagun74** [reply]
yes, what exactly do you want to achieve? Maybe there is a smarter way.

**Reply by wieluk** [reply]
save game sync with ludusavi. There is a plugin for playnite that does exactly this. I know some safe file management is planned but I wanted to try if i can get it to work.
If it is done in gamevault I wouldn't mind to pay a onetime payment for it but I don't want a subscritption ...

**Reply by Alfagun74** [reply]
I totally respect your viewpoint on the subscription model, and I hope you understand our need for a steady income as a new company aswell. You can work your way around this, like with this script if you dont want to support us regurarly. We dont mind at all, its up to you.

Regarding the script, I would suggest writing it once, then copying it into each game folder as needed. This script should automatically run and keep track of the first available executable file in the folder by default. Additionally, it could prompt the user to select a executable if there are multiple options. Then write your own gamevault-exec file to persist that option. 


Or, if you're familiar with programming in .NET, you could edit the code in the GameTimeTracker.cs file found here: https://github.com/Phalcode/gamevault-app/blob/master/gamevault/Helper/GameTimeTracker.cs

**Comment by Yelo420** [comment]
Gamevault can also execute all scripts that can be executed on Windows. BAT, VBS and so on...

You could also cover the execution of a script after the game has ended by waiting in the start script for the game process to end.
